Sources/SPMBuildCore/Plugins/PluginInvocation.swift

```diff
@@ -485,7 +488,9 @@ extension PackageGraph {
                         dispatchPrecondition(condition: .onQueue(delegateQueue))
                         outputData.append(contentsOf: data)
                     }
-                    
+
+                    func pluginEmittedProgress(_ message: String) {}
```

Comment:
Not clear what is the best option here.   Could append to `outputData` or add a `progress` array field similar to `diagnostics` below.